static/app/components/hook.tsx

```diff
@@ -51,7 +53,9 @@ function Hook<H extends HookName>({name, ...props}: Props<H>) {
         return;
       }
 
-      this.setState({hooks: hooks.map(cb => cb(props))});
```

Comment:
is it always a compoenent?

Reply:
Hmm I figured that the point of this was to render getsentry components so I hope so. Maybe I can get the types to validate that

Reply:
Okay added some type checking, looks like all the current usages are indeed components.